service/src/test/java/org/sunbird/learner/actors/UserFrameworkUpdateTest.java

```diff
@@ -0,0 +1,212 @@
+
+    try {
+      PowerMockito.whenNew(UserExternalIdentityDaoImpl.class)
+          .withNoArguments()
+          .thenReturn(userExtDao);
+
+    } catch (Exception e) {
+      ProjectLogger.log("Error while mocking UserExternalIdentityDaoImpl");
+    }
+  }
+
+  private Request getRequest(String key, String value) {
+    Request reqObj = new Request();
+    reqObj.setOperation(ActorOperations.UPDATE_USER.getValue());
+    Map<String, Object> innerMap = new HashMap<>();
+    innerMap.put(JsonKey.ID, userId);
+    Map<String, Object> frameworkMap = getFrameworkDetails(key, value);
+
+    innerMap.put(JsonKey.FRAMEWORK, frameworkMap);
+    Map<String, Object> request = new HashMap<String, Object>();
+    request.put(JsonKey.USER, innerMap);
+    request.put(JsonKey.USER_ID, userId);
+    request.put(JsonKey.FRAMEWORK, frameworkMap);
+    reqObj.setRequest(request);
+    Map<String, Object> context = new HashMap<>();
+    context.put(JsonKey.REQUESTED_BY, "someValue");
+    context.put(JsonKey.USER_ID, userId);
+    reqObj.setContext(context);
+    return reqObj;
+  }
+
+  private Map<String, Object> getFrameworkDetails(String key, String value) {
+    Map<String, Object> frameworkMap = new HashMap<>();
+    List<String> medium = new ArrayList<>();
+    medium.add("English");
+    List<String> gradeLevel = new ArrayList<>();
+    gradeLevel.add("Grade 3");
+    List<String> board = new ArrayList<>();
+    board.add("NCERT");
+    frameworkMap.put(JsonKey.ID, "NCF");
+    frameworkMap.put("gradeLevel", gradeLevel);
+    frameworkMap.put("board", board);
+    frameworkMap.put("medium", medium);
+    if (key != null) {
+      List<String> wrongValue = new ArrayList<>();
+      wrongValue.add(value);
+      frameworkMap.put(key, wrongValue);
+    }
+    return frameworkMap;
+  }
+
+  private void mockuserExtDao(Request req) {
+    when(userExtDao.getUserId(req)).thenReturn(userId);
+  }
+
+  private void mockUtilsForOrgDetails() {
+    Map<String, Object> rootOrgMap = new HashMap<>();
+    String hashTagId = "someHashTagId";
+    rootOrgMap.put(JsonKey.HASHTAGID, hashTagId);
+    when(Util.getOrgDetails(Mockito.anyString())).thenReturn(rootOrgMap);
+  }
+
+  private void mockContentStoreUtil() {
+    Map<String, Object> contentMap = new HashMap<>();
+    contentMap.put(JsonKey.RESPONSE, null);
+    when(ContentStoreUtil.readFramework("invalidFrameworkId")).thenReturn(contentMap);
+  }
+
+  private void mockDatacacheHandler() {
+    Map<String, List<String>> frameworkFieldsConfigMap = new HashMap<>();
+    List<String> frameworkFieldConfig =
+        Arrays.asList("id", "medium", "gradeLevel", "board", "subject");
+    List<String> frameworkFieldConfigMan = Arrays.asList("id", "medium", "gradeLevel", "board");
+    frameworkFieldsConfigMap.put(JsonKey.FIELDS, frameworkFieldConfig);
+    frameworkFieldsConfigMap.put(JsonKey.MANDATORY_FIELDS, frameworkFieldConfigMan);
+    DataCacheHandler.setFrameworkFieldsConfig(frameworkFieldsConfigMap);
```

Comment:
u don't need to use setters, just mock and return the wishful results from functions that return values.